compiler/codegen/symbol.cpp

```diff
@@ -1204,6 +1222,9 @@ void FnSymbol::codegenDef() {
 #ifdef HAVE_LLVM
     func = getFunctionLLVM(cname);
 
+    if(strcmp(llvmFuncDumpName, name) == 0)
+        llvmFuncDumpCName = strdup(cname);
```

Comment:
Because of the way the Chapel compiler manages strings, you don't need to call strdup here. Just make it point to cname.